ui/src/types/stores.ts

```diff
@@ -40,6 +43,7 @@ export interface AppState {
   }
   currentPage: CurrentPage
   currentDashboard: CurrentDashboardState
+  data: DataState
```

Comment:
i'm going to be blunt here: `data` is a really poor name for a reducer - all reducers deal in data. i'm not sure if this reducer needs to exist, but if it does, what's wrong with calling it the `cache` reducer or the `queryCache` reducer?

Reply:
Those are really great names, I'll switch them up